ReactiveCocoa/Swift/Property.swift

```diff
@@ -97,16 +108,27 @@ public final class MutableProperty<Value>: MutablePropertyType {
 		}
 	}
 
+	/// A signal that will send the property's changes over time,
+	/// then complete when the property has deinitialized.
+	public let signal: Signal<Value, NoError>
+
 	/// A producer for Signals that will send the property's current value,
 	/// followed by all changes over time, then complete when the property has
 	/// deinitialized.
-	public let producer: SignalProducer<Value, NoError>
+	public var producer: SignalProducer<Value, NoError> {
+		return SignalProducer { observer, disposable in
+			self.withValue {
```

Comment:
I'm not sure about this: isn't this an unnecessary ownership? As long as there's a copy of this producer the property will be retained, which changes the semantics.

I still think that the implementation would be more correct by keeping the implementation using `SignalProducer.buffer` and using `.skip(1)` for the signal.